### ISSUE TITLE
Reverse Proxy not working

### ISSUE THREAD

**Issue by patrickjmccarty** [jippi/hashi-ui#420]
I tried to setup hashi-ui reverse-proxied behind nginx, but this capability seems to be broken in hashi-ui 1.0.0. I'm a new user, so I have no idea when this last worked. I was hoping to make Hashi-UI available at a url like: http://hostname/customprefix

I tried a bunch of things, thinking it must be my fault since nginx and reverse proxying were mentioned in various places implying that this should be supported. But after looking at the developer console and the html that is served I'm convinced that this is a bug.

HASHI_ASSETS_ROOT seems to be at least one part of the problem. The homepage source tries to use this variable to build a script url, but it is not defined. Even when not using a reverse proxy, this actually results in it loading a url that contains 'undefined' like: http://127.0.0.1:3000/undefined/static/app.fa434117136573a3c3d0.js
It seems really weird that the 'undefined' url actually works, when it seems that it should have been http://127.0.0.1:3000/static/app.fa434117136573a3c3d0.js instead (which also works). Regardless, numerous things seem to be wrong so I'm not going to try to analyze this further.

**Comment by jippi** [comment]
[have you tried the scripts example?](https://github.com/jippi/hashi-ui/tree/master/scripts/nginx-frontend-proxy)

**Comment by patrickjmccarty** [comment]
Ok, I finally got it working after carefully following that example and using the argument --proxy-address /customprefix
I was thinking the --proxy-address would mean I could do it without a trailing slash on the proxy_pass directive, since that's how most other things I've reverse-proxied work, but that was wrong in this case. Adding to my confusion was that hashi-ui responds to any and all incorrect urls (for ex: http://127.0.0.1:3000/asdflakjsdf/asdaslkj) rather than returning 404, which makes it hard to check that you are on the right track. So I was seeing code 200 responses with curl and thinking I had the proxy_pass correct, but javascript things weren't working in the browser.

I still have one issue that could be considered a bug:
The links on the left navigation sidebar work fine when clicked, but appear wrong in the statusbar (they are missing my /customprefix), and so they do not work if you right-click and 'Copy link address', or if you 'Open link in new tab'.
I added the following lame workaround to my nginx config to deal with this for now, but this should be fixed in the application.
    location /nomad/global {
        return 301 http://$host/customprefix$request_uri;
    }
Thanks!
